modules/signals/src/deep-signal.ts

```diff
@@ -1,6 +1,18 @@
-import { computed, Signal, untracked } from '@angular/core';
+import {
+  computed,
+  isSignal,
+  Signal as NgSignal,
+  untracked,
+} from '@angular/core';
 import { IsUnknownRecord } from './ts-helpers';
 
+// An extended Signal type that enables the correct typing
+// of nested signals with the `name' or `length' key.
+interface Signal<T> extends NgSignal<T> {
```

Comment:
I think creator our own type with the same name can be confusing.
Could we rename this to `ExtendedSignal` or something else?

Reply:
If we rename it, IDE will show `ExtendedSignal` instead of `Signal` for SignalState/Store properties, which will be confusing to users. Extended signal type is private and fully compatible with Angular's Signal type.

There is no other way to provide correct typing if a SignalState/Store contains `name` or `length` properties (at least I didn't find it). If we do: `Omit<Signal<T>, 'name' | 'length'>, TypeScript will not recognize that type as a function, so it won't be executable anymore.

Reply:
I see, I thought this was just for internal use.